# Etude de plusieurs facteurs environnementaux sur la production dans les hôpitaux

In [86]:
# importation des packages
import os 
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
import statsmodels.api as sm

In [87]:
VAR_CTRL = pd.read_csv('https://raw.githubusercontent.com/VincianeDesbois/Hopitaux_Production/main/Variables_controle.csv',  sep = ',')
VAR_EXPL_CAT  = pd.read_csv('https://raw.githubusercontent.com/VincianeDesbois/Hopitaux_Production/main/VAR_EXPL_CAT.csv',  sep = ',')

In [88]:
VAR_EXPL_CAT

Unnamed: 0 Numero_finess  Médecins libéraux  \
0              1     010008407                0.0   
1              2     010009132                0.0   
2              3     010780054                4.5   
3              4     010780062                0.0   
4              5     010780096                0.0   
...          ...           ...                ...   
1147        1117     940000649                0.0   
1148        1118     940000656                0.0   
1149        1119     940000664                0.0   
1150        1135     950000406                0.0   
1151        1150     950630012                0.0   

      Autres personnels médicaux libéraux  Médecins salariés  \
0                                     0.0             18.490   
1                                     0.0              1.795   
2                                     1.0            102.600   
3                                     0.0             19.450   
4                                     0.0             10.260   
...                                   ...                ...   
1147                                  0.0             97.680   
1148                                  0.0              9.460   
1149                                  0.0            202.110   
1150                                  0.0              3.510   
1151                                  0.0             11.300   

      Autres personnels médicaux salariés  \
0                                  14.330   
1                                   0.070   
2                                  44.530   
3                                  10.100   
4                                   1.885   
...                                   ...   
1147                               28.690   
1148                                1.000   
1149                               79.550   
1150                                1.260   
1151                                2.500   

      Personnels de direction et administratifs  \
0                                     21.196667   
1                                      4.175000   
2                                    118.605000   
3                                     58.450000   
4                                      9.685000   
...                                         ...   
1147                                 132.300000   
1148                                  39.470000   
1149                                 459.580000   
1150                                  13.950000   
1151                                  23.070000   

      Personnels des services de soins  Personnels éducatifs et sociaux  \
0                            97.323333                         0.846667   
1                            46.700000                         1.065000   
2                           388.590000                         4.930000   
3                           336.540000                         5.450000   
4                            88.475000                         2.555000   
...                                ...                              ...   
1147                        468.280000                         4.830000   
1148                         93.680000                         2.520000   
1149                        904.180000                         9.700000   
1150                         78.740000                         1.980000   
1151                        177.880000                        39.070000   

      Personnels médico-techniques  Personnels techniques et ouvriers  \
0                         9.206667                             26.440   
1                         1.130000                             13.465   
2                       107.660000                            112.630   
3                        29.320000                             57.570   
4                         2.570000                              5.960   
...                            ...                                ...   
1147                     58.750000

# Première analyse sur l'effet du revenu fiscal moyen sur les points ISA

In [89]:
VAR_EXPL_CAT_1 = VAR_EXPL_CAT[['Numero_finess', 'Personnel médical','STJR_1','STJR_2', 'STJR_3',
       'Personnels de direction et administratifs',
       'Personnels des services de soins', 'Personnels éducatifs et sociaux',
       'Personnels médico-techniques', 'Personnels techniques et ouvriers',
        'Internes']]
VAR_EXPL_CAT_1= VAR_EXPL_CAT_1.fillna(0)

In [90]:
VAR_EXPL_CAT_1

Numero_finess  Personnel médical  STJR_1  STJR_2  STJR_3  \
0        010008407             32.820     1.0     0.0     0.0   
1        010009132              1.865     1.0     0.0     0.0   
2        010780054            152.630     1.0     0.0     0.0   
3        010780062             29.550     1.0     0.0     0.0   
4        010780096             12.145     1.0     0.0     0.0   
...            ...                ...     ...     ...     ...   
1147     940000649            126.370     0.0     0.0     1.0   
1148     940000656             10.460     0.0     0.0     1.0   
1149     940000664            281.660     0.0     0.0     1.0   
1150     950000406              4.770     0.0     0.0     1.0   
1151     950630012             13.800     0.0     0.0     1.0   

      Personnels de direction et administratifs  \
0                                     21.196667   
1                                      4.175000   
2                                    118.605000   
3                                     58.450000   
4                                      9.685000   
...                                         ...   
1147                                 132.300000   
1148                                  39.470000   
1149                                 459.580000   
1150                                  13.950000   
1151                                  23.070000   

      Personnels des services de soins  Personnels éducatifs et sociaux  \
0                            97.323333                         0.846667   
1                            46.700000                         1.065000   
2                           388.590000                         4.930000   
3                           336.540000                         5.450000   
4                            88.475000                         2.555000   
...                                ...                              ...   
1147                        468.280000                         4.830000   
1148                         93.680000                         2.520000   
1149                        904.180000                         9.700000   
1150                         78.740000                         1.980000   
1151                        177.880000                        39.070000   

      Personnels médico-techniques  Personnels techniques et ouvriers  \
0                         9.206667                             26.440   
1                         1.130000                             13.465   
2                       107.660000                            112.630   
3                        29.320000                             57.570   
4                         2.570000                              5.960   
...                            ...                                ...   
1147                     58.750000                             25.760   
1148                      1.060000                             51.490   
1149                    404.530000                            126.770   
1150                      5.580000                              2.360   
1151                      5.330000                             12.700   

      Internes  
0         0.00  
1         0.00  
2       195.09  
3         6.45  
4         3.48  
...        ...  
1147    132.00  
1148      1.47  
1149    402.00  
1150      0.00  
1151      2.49  

[1152 rows x 11 columns]

In [91]:
VAR_EXPL_CAT_1 = VAR_EXPL_CAT_1.apply(lambda x: x+1 if x.name in ['Personnel médical',
       'Personnels de direction et administratifs',
       'Personnels des services de soins', 'Personnels éducatifs et sociaux',
       'Personnels médico-techniques', 'Personnels techniques et ouvriers',
        'Internes'] else x)


VAR_EXPL_CAT_1 = VAR_EXPL_CAT_1.apply(lambda x: np.log(x) if x.name in ['Personnel médical',
       'Personnels de direction et administratifs',
       'Personnels des services de soins', 'Personnels éducatifs et sociaux',
       'Personnels médico-techniques', 'Personnels techniques et ouvriers',
        'Internes'] else x)

In [92]:
VAR_EXPL_CAT_1

Numero_finess  Personnel médical  STJR_1  STJR_2  STJR_3  \
0        010008407           3.521052     1.0     0.0     0.0   
1        010009132           1.052568     1.0     0.0     0.0   
2        010780054           5.034547     1.0     0.0     0.0   
3        010780062           3.419365     1.0     0.0     0.0   
4        010780096           2.576041     1.0     0.0     0.0   
...            ...                ...     ...     ...     ...   
1147     940000649           4.847096     0.0     0.0     1.0   
1148     940000656           2.438863     0.0     0.0     1.0   
1149     940000664           5.644245     0.0     0.0     1.0   
1150     950000406           1.752672     0.0     0.0     1.0   
1151     950630012           2.694627     0.0     0.0     1.0   

      Personnels de direction et administratifs  \
0                                      3.099942   
1                                      1.643839   
2                                      4.784195   
3                                      4.085136   
4                                      2.368841   
...                                         ...   
1147                                   4.892602   
1148                                   3.700561   
1149                                   6.132487   
1150                                   2.704711   
1151                                   3.180966   

      Personnels des services de soins  Personnels éducatifs et sociaux  \
0                             4.588261                         0.613382   
1                             3.864931                         0.725130   
2                             5.965095                         1.780024   
3                             5.821684                         1.864080   
4                             4.493959                         1.268355   
...                                ...                              ...   
1147                          6.151200                         1.763017   
1148                          4.550503                         1.258461   
1149                          6.808134                         2.370244   
1150                          4.378771                         1.091923   
1151                          5.186715                         3.690628   

      Personnels médico-techniques  Personnels techniques et ouvriers  \
0                         2.323041                           3.312002   
1                         0.756122                           2.671732   
2                         4.688224                           4.732948   
3                         3.411808                           4.070223   
4                         1.272566                           1.940179   
...                            ...                                ...   
1147                      4.090169                           3.286908   
1148                      0.722706                           3.960623   
1149                      6.005195                           4.850232   
1150                      1.884035                           1.211941   
1151                      1.845300                           2.617396   

      Internes  
0     0.000000  
1     0.000000  
2     5.278574  
3     2.008214  
4     1.499623  
...        ...  
1147  4.890349  
1148  0.904218  
1149  5.998937  
1150  0.000000  
1151  1.249902  

[1152 rows x 11 columns]

In [93]:
VAR_CTRL.columns
#Je prends Cpaital_LIT_MCO et Capital_Bloc

Index(['Numero_finess', 'Points_ISA', 'stjr', 'dep', 'Departement', 'COMINSEE',
       'Capital_Lit_MCO', 'Capital_Bloc', 'Capital_imagerie',
       'Capital_Examens_biologiques', 'Capital_telemed', 'SEJHC_MED',
       'SEJHC_CHI', 'SEJHC_OBS', 'SEJHC_MCO', 'Taux_chomage_par_dep',
       'Population', 'Dep Moyenne Salaires Horaires',
       'Dep Moyenne Salaires Cadre Horaires',
       'Dep Moyenne Salaires Prof Intermédiaire Horaires',
       'Dep Moyenne Salaires Employé Horaires',
       'Moyenne Revenus Fiscaux Départementaux', 'Bloc_politique'],
      dtype='object')

In [94]:
VAR_CTRL_1 = VAR_CTRL.copy()
VAR_CTRL_1 = VAR_CTRL_1.fillna(0)
VAR_CTRL_1 = VAR_CTRL_1.apply(lambda x: x+1 if x.name in ['Points_ISA', 'Capital_Lit_MCO', 'Capital_Bloc', 'Capital_imagerie',
       'Capital_Examens_biologiques', 'Capital_telemed', 'Population'] else x)

VAR_CTRL_1 = VAR_CTRL_1.apply(lambda x: np.log(x) if x.name in ['Points_ISA', 'Capital_Lit_MCO', 'Capital_Bloc', 'Capital_imagerie',
       'Capital_Examens_biologiques', 'Capital_telemed', 'Population'] else x)

In [95]:
VAR_CTRL.describe()

Points_ISA         stjr  Capital_Lit_MCO  Capital_Bloc  \
count  1.162000e+03  1162.000000      1139.000000   1153.000000   
mean   2.939915e+05     1.707401       125.109263      7.379577   
std    6.281782e+05     0.716795       127.684194      7.191520   
min    1.347900e+02     1.000000         0.000000      0.000000   
25%    2.993899e+04     1.000000        35.000000      1.000000   
50%    1.016205e+05     2.000000        85.000000      6.000000   
75%    2.975944e+05     2.000000       171.166667     10.500000   
max    9.116903e+06     3.000000      1055.000000     44.000000   

       Capital_imagerie  Capital_Examens_biologiques  Capital_telemed  \
count       1162.000000                 1.149000e+03      1162.000000   
mean           4.256971                 2.508881e+05         0.728148   
std            5.047103                 6.567670e+05         1.514649   
min            0.000000                 0.000000e+00         0.000000   
25%            0.000000                 0.000000e+00         0.000000   
50%            3.000000                 0.000000e+00         0.000000   
75%            7.000000                 6.261400e+04         1.000000   
max           28.000000                 1.060114e+07        11.000000   

          SEJHC_MED     SEJHC_CHI    SEJHC_OBS     SEJHC_MCO  \
count   1162.000000   1162.000000  1161.000000   1162.000000   
mean    3953.236728   2091.528256   673.093714   6717.279443   
std     5172.505167   2471.292213  1146.876402   7600.700172   
min        0.000000      0.000000     0.000000      0.000000   
25%      364.750000      0.000000     0.000000   1204.500000   
50%     1963.000000   1428.000000     0.000000   4168.500000   
75%     5581.250000   3168.312500  1081.000000   9749.750000   
max    39884.000000  20991.000000  7639.000000  60719.000000   

       Taux_chomage_par_dep    Population  Dep Moyenne Salaires Horaires  \
count           1162.000000  1.162000e+03                    1162.000000   
mean               8.290103  1.367009e+05                      11.793749   
std                1.660836  3.998901e+05                       1.908702   
min                5.000000  1.820000e+02                       9.862500   
25%                7.000000  8.053750e+03                      10.633333   
50%                8.000000  2.295500e+04                      11.204762   
75%                9.300000  6.171825e+04                      11.981429   
max               13.800000  2.125851e+06                      17.800000   

       Dep Moyenne Salaires Cadre Horaires  \
count                          1162.000000   
mean                             21.764363   
std                               1.674054   
min                              19.750000   
25%                              20.775000   
50%                              21.280808   
75%                              22.083750   
max                              27.600000   

       Dep Moyenne Salaires Prof Intermédiaire Horaires  \
count                                       1162.000000   
mean                                          12.674821   
std                                            0.523229   
min                                           11.740000   
25%                                           12.346721   
50%                                           12.535714   
75%                                           12.902500   
max                                           14.185000   

       Dep Moyenne Salaires Employé Horaires  \
count                            1162.000000   
mean                                8.783985   
std                                 0.353571   
min                                 8.261905   
25%                                 8.555172   
50%                                 8.700862   
75%                                 8.848113   
max                                 9.742553   

       Moyenne Revenus Fiscaux Départementaux  
count                             1162.000000  
mean     

Le premier quartile des revenus fiscaux moyens par mois est à 9471, et le troisième quartile à 11536. On sépare en deux bases : celles avec les départements avec les plus faibles revenus, et celles avec les départements avec les plus hauts revenus.

In [96]:
# Les établissements avec des revenus départements  trés élevés

VAR_CTRL_ELEVES = VAR_CTRL_1.copy()
VAR_CTRL_ELEVES= VAR_CTRL_ELEVES.rename(columns = {'Moyenne Revenus Fiscaux Départementaux': 'Rev_dep'})

VAR_CTRL_ELEVES= VAR_CTRL_ELEVES[VAR_CTRL_ELEVES.Rev_dep > 10239]

In [97]:
# Les établissements avec des revenus départements trésfaibles

VAR_CTRL_FAIBLES = VAR_CTRL_1.copy()
VAR_CTRL_FAIBLES = VAR_CTRL_FAIBLES.rename(columns = {'Moyenne Revenus Fiscaux Départementaux': 'Rev_dep'})

VAR_CTRL_FAIBLES = VAR_CTRL_FAIBLES[VAR_CTRL_FAIBLES.Rev_dep <= 10239]

In [98]:
VAR_CTRL_FAIBLES

Numero_finess  Points_ISA  stjr dep       Departement COMINSEE  \
20       130001647   13.658610     3  13  Bouches-du-Rhône    13055   
21       130784051   12.784335     2  13  Bouches-du-Rhône    13055   
22       130783152    8.674370     3  13  Bouches-du-Rhône    13055   
23       130783665   11.152202     3  13  Bouches-du-Rhône    13055   
24       130785652   14.297203     3  13  Bouches-du-Rhône    13055   
...            ...         ...   ...  ..               ...      ...   
1157     120780085   10.729125     1  12           Aveyron    12089   
1158     480780097   12.391989     1  48            Lozère    48095   
1159     480780162    8.648248     1  48            Lozère    48080   
1160     480780121    7.782161     1  48            Lozère    48140   
1161     480780154    6.251287     1  48            Lozère    48092   

      Capital_Lit_MCO  Capital_Bloc  Capital_imagerie  \
20           5.605802      2.564949          2.639057   
21           5.693732      2.995732          2.639057   
22           2.564949      0.000000          0.000000   
23           3.912023      1.791759          1.945910   
24           6.520621      3.401197          2.708050   
...               ...           ...               ...   
1157         3.555348      1.791759          1.386294   
1158         4.219508      1.704748          1.945910   
1159         3.178054      0.000000          0.693147   
1160         3.135494      0.000000          0.000000   
1161         3.044522      0.000000          0.000000   

      Capital_Examens_biologiques  ...  SEJHC_OBS  SEJHC_MCO  \
20                       9.733351  ...        0.0    15228.0   
21                       0.000000  ...        0.0    12231.0   
22                       0.000000  ...        0.0      419.0   
23                      11.171055  ...        0.0     1525.0   
24                      12.740877  ...     6414.0    40357.0   
...                           ...  ...        ...        ...   
1157                    12.367634  ...        0.0     2056.0   
1158                    12.956160  ...      392.0     3599.5   
1159                     0.000000  ...        0.0      421.0   
1160                     0.000000  ...        0.0      205.0   
1161                     0.000000  ...        0.0      384.0   

      Taux_chomage_par_dep  Population  Dep Moyenne Salaires Horaires  \
20                    10.1   13.589227                      12.477895   
21                    10.1   13.589227                      12.477895   
22                    10.1   13.589227                      12.477895   
23                    10.1   13.589227                      12.477895   
24                    10.1   13.589227                      12.477895   
...                    ...         ...                            ...   
1157                   6.4    8.822912                      10.115789   
1158                   5.3    9.375261                      10.075000   
1159                   5.3    8.038512                      10.075000   
1160                   5.3    8.371011                      10.075000   
1161                   5.3    8.613412                      10.075000   

      Dep Moyenne Salaires Cadre Horaires  \
20                              22.122105   
21                              22.122105   
22                              22.122105   
23                              22.122105   
24                              22.122105   
...                                   ...   
1157                            20.252632   
1158                            20.475000   
1159                            20.475000   
1160                            20.475000   
1161                            20.475000   

      Dep Moyenne Salaires Prof Intermédiaire Horaires  \
20                                           13.044211   
21                                           13.044211   
22                                           13.044211   
23                                           13.

In [99]:
analyse_dep_faibles = pd.merge(VAR_CTRL_FAIBLES, VAR_EXPL_CAT_1)

analyse_dep_eleves = pd.merge(VAR_CTRL_ELEVES, VAR_EXPL_CAT_1)

In [100]:
analyse_dep_faibles.columns

Index(['Numero_finess', 'Points_ISA', 'stjr', 'dep', 'Departement', 'COMINSEE',
       'Capital_Lit_MCO', 'Capital_Bloc', 'Capital_imagerie',
       'Capital_Examens_biologiques', 'Capital_telemed', 'SEJHC_MED',
       'SEJHC_CHI', 'SEJHC_OBS', 'SEJHC_MCO', 'Taux_chomage_par_dep',
       'Population', 'Dep Moyenne Salaires Horaires',
       'Dep Moyenne Salaires Cadre Horaires',
       'Dep Moyenne Salaires Prof Intermédiaire Horaires',
       'Dep Moyenne Salaires Employé Horaires', 'Rev_dep', 'Bloc_politique',
       'Personnel médical', 'STJR_1', 'STJR_2', 'STJR_3',
       'Personnels de direction et administratifs',
       'Personnels des services de soins', 'Personnels éducatifs et sociaux',
       'Personnels médico-techniques', 'Personnels techniques et ouvriers',
       'Internes'],
      dtype='object')

In [101]:
analyse_dep_faibles_X =analyse_dep_faibles[["Personnel médical","Internes",'Personnels de direction et administratifs',
       'Personnels des services de soins', 'Personnels éducatifs et sociaux',
       'Personnels médico-techniques', 'Personnels techniques et ouvriers', 
        'Capital_Lit_MCO', 'Capital_Bloc', 'Capital_imagerie', 'Population',
       'Capital_Examens_biologiques', 'Capital_telemed', 'STJR_1', 'STJR_2', 'STJR_3']]
analyse_dep_faibles_Y =analyse_dep_faibles[['Points_ISA']]
analyse_dep_eleves_X =analyse_dep_eleves[["Personnel médical","Internes",'Personnels de direction et administratifs',
       'Personnels des services de soins', 'Personnels éducatifs et sociaux',
       'Personnels médico-techniques', 'Personnels techniques et ouvriers', 
       'Capital_Lit_MCO', 'Capital_Bloc', 'Capital_imagerie', 'Population',
       'Capital_Examens_biologiques', 'Capital_telemed', 'STJR_1', 'STJR_2', 'STJR_3']]
analyse_dep_eleves_Y =analyse_dep_eleves[['Points_ISA']]

In [102]:
analyse_dep_faibles_X 

Personnel médical  Internes  Personnels de direction et administratifs  \
0             5.254574  5.503582                                   5.542518   
1             4.485373  0.000000                                   3.832547   
2             1.968510  0.000000                                   2.701361   
3             4.060271  0.000000                                   3.284664   
4             5.451896  4.007333                                   5.753144   
..                 ...       ...                                        ...   
575           3.270709  1.081805                                   3.443618   
576           4.086228  2.944439                                   3.568593   
577           1.589235  0.000000                                   2.079442   
578           2.047693  0.000000                                   1.015231   
579           1.611436  0.000000                                   1.501853   

     Personnels des services de soins  Personnels éducatifs et sociaux  \
0                            6.437608                         2.830858   
1                            6.139475                         0.693147   
2                            4.027314                         0.693147   
3                            4.392843                         0.693147   
4                            7.219788                         2.046402   
..                                ...                              ...   
575                          5.170768                         1.098612   
576                          5.004617                         1.688249   
577                          3.858622                         0.693147   
578                          3.191299                         0.000000   
579                          3.677313                         0.165514   

     Personnels médico-techniques  Personnels techniques et ouvriers  \
0                        5.618987                           4.548282   
1                        2.832036                           3.212455   
2                        0.182322                           1.252763   
3                        2.341806                           2.564949   
4                        4.887262                           4.843084   
..                            ...                                ...   
575                      3.430756                           3.658420   
576                      2.976549                           3.140482   
577                      0.693147                           2.505526   
578                      0.223144                           1.560248   
579                      0.000000                           2.054124   

     Capital_Lit_MCO  Capital_Bloc  Capital_imagerie  Population  \
0           5.605802      2.564949          2.639057   13.589227   
1           5.693732      2.995732          2.639057   13.589227   
2           2.564949      0.000000          0.000000   13.589227   
3           3.912023      1.791759          1.945910   13.589227   
4           6.520621      3.401197          2.708050   13.589227   
..               ...           ...               ...         ...   
575         3.555348      1.791759          1.386294    8.822912   
576         4.219508      1.704748          1.945910    9.375261   
577         3.178054      0.000000          0.693147    8.038512   
578         3.135494      0.000000          0.000000    8.371011   
579         3.044522      0.000000          0.000000    8.613412   

     Capital_Examens_biologiques  Capital_telemed  STJR_1  STJR_2  STJR_3  
0                       9.733351         0.000000     0.0     0.0     1.0  
1                       0.000000         0.000000     0.0     1.0     0.0  
2                       0.000000         0.693147     0.0     0.0     1.0  
3                      11.171055         0.000000     0.0     0.0     1.0  
4                      12.740877         0.000000     0.0     0.0     1.0  
..                           

In [103]:
analyse_dep_faibles_X = sm.add_constant(analyse_dep_faibles_X) # on ajoute une constante, sinon la régression se fait sans, par défaut
y = analyse_dep_faibles[['Points_ISA']]

model_dep_faibles = sm.OLS(y, analyse_dep_faibles_X)
results_dep_faibles = model_dep_faibles.fit()
print(results_dep_faibles.summary())

                            OLS Regression Results                            
Dep. Variable:             Points_ISA   R-squared:                       0.907
Model:                            OLS   Adj. R-squared:                  0.904
Method:                 Least Squares   F-statistic:                     365.1
Date:                Wed, 19 May 2021   Prob (F-statistic):          1.04e-278
Time:                        15:35:23   Log-Likelihood:                -462.05
No. Observations:                 580   AIC:                             956.1
Df Residuals:                     564   BIC:                             1026.
Df Model:                          15                                         
Covariance Type:            nonrobust                                         
                                                coef    std err          t      P>|t|      [0.025      0.975]
----------------------------------------------------------------------------------------------------

In [104]:
analyse_dep_eleves_X = sm.add_constant(analyse_dep_eleves_X) # on ajoute une constante, sinon la régression se fait sans, par défaut
y = analyse_dep_eleves[['Points_ISA']]

model_dep_eleves = sm.OLS(y, analyse_dep_eleves_X)
results_dep_eleves = model_dep_eleves.fit()
print(results_dep_eleves.summary())

                            OLS Regression Results                            
Dep. Variable:             Points_ISA   R-squared:                       0.886
Model:                            OLS   Adj. R-squared:                  0.883
Method:                 Least Squares   F-statistic:                     289.2
Date:                Wed, 19 May 2021   Prob (F-statistic):          5.20e-251
Time:                        15:35:23   Log-Likelihood:                -508.39
No. Observations:                 572   AIC:                             1049.
Df Residuals:                     556   BIC:                             1118.
Df Model:                          15                                         
Covariance Type:            nonrobust                                         
                                                coef    std err          t      P>|t|      [0.025      0.975]
----------------------------------------------------------------------------------------------------

# Deuxième analyse : effet de la moyenne des salaires des cadres sur l'emploi de personnel médical libéral

In [105]:
VAR_EXPL_CAT_2 = VAR_EXPL_CAT.copy()

In [106]:
VAR_EXPL_CAT_2['Personnel médical salarié'] = VAR_EXPL_CAT['Médecins salariés'] + VAR_EXPL_CAT["Autres personnels médicaux salariés"]
VAR_EXPL_CAT_2['Personnel médical libéral'] = VAR_EXPL_CAT['Médecins libéraux'] + VAR_EXPL_CAT["Autres personnels médicaux libéraux"]

In [107]:
VAR_EXPL_CAT_2 = VAR_EXPL_CAT_2[['Numero_finess', 'Personnel médical salarié','Personnel médical libéral','STJR_1', 'STJR_2', 'STJR_3',
       'Personnels de direction et administratifs',
       'Personnels des services de soins', 'Personnels éducatifs et sociaux',
       'Personnels médico-techniques', 'Personnels techniques et ouvriers',
        'Internes']]

In [108]:
VAR_EXPL_CAT_2

Numero_finess  Personnel médical salarié  Personnel médical libéral  \
0        010008407                     32.820                        0.0   
1        010009132                      1.865                        0.0   
2        010780054                    147.130                        5.5   
3        010780062                     29.550                        0.0   
4        010780096                     12.145                        0.0   
...            ...                        ...                        ...   
1147     940000649                    126.370                        0.0   
1148     940000656                     10.460                        0.0   
1149     940000664                    281.660                        0.0   
1150     950000406                      4.770                        0.0   
1151     950630012                     13.800                        0.0   

      STJR_1  STJR_2  STJR_3  Personnels de direction et administratifs  \
0        1.0     NaN     NaN                                  21.196667   
1        1.0     NaN     NaN                                   4.175000   
2        1.0     NaN     NaN                                 118.605000   
3        1.0     NaN     NaN                                  58.450000   
4        1.0     NaN     NaN                                   9.685000   
...      ...     ...     ...                                        ...   
1147     NaN     NaN     1.0                                 132.300000   
1148     NaN     NaN     1.0                                  39.470000   
1149     NaN     NaN     1.0                                 459.580000   
1150     NaN     NaN     1.0                                  13.950000   
1151     NaN     NaN     1.0                                  23.070000   

      Personnels des services de soins  Personnels éducatifs et sociaux  \
0                            97.323333                         0.846667   
1                            46.700000                         1.065000   
2                           388.590000                         4.930000   
3                           336.540000                         5.450000   
4                            88.475000                         2.555000   
...                                ...                              ...   
1147                        468.280000                         4.830000   
1148                         93.680000                         2.520000   
1149                        904.180000                         9.700000   
1150                         78.740000                         1.980000   
1151                        177.880000                        39.070000   

      Personnels médico-techniques  Personnels techniques et ouvriers  \
0                         9.206667                             26.440   
1                         1.130000                             13.465   
2                       107.660000                            112.630   
3                        29.320000                             57.570   
4                         2.570000                              5.960   
...                            ...                                ...   
1147                     58.750000                             25.760   
1148                      1.060000                             51.490   
1149                    404.530000                            126.770   
1150                      5.580000                              2.360   
1151                      5.330000                             12.700   

      Internes  
0         0.00  
1         0.00  
2       195.09  
3         6.45  
4         3.48  
...        ...  
1147    132.00  
1148      1.47  
1149    402.00  
1150      0.00  
1151      2.49  

[1152 rows x 12 columns]

In [109]:
VAR_EXPL_CAT_2 = VAR_EXPL_CAT_2.fillna(0)

VAR_EXPL_CAT_2 = VAR_EXPL_CAT_2.apply(lambda x: x+1 if x.name in ['Personnel médical salarié','Personnel médical libéral',
       'Personnels de direction et administratifs',
       'Personnels des services de soins', 'Personnels éducatifs et sociaux',
       'Personnels médico-techniques', 'Personnels techniques et ouvriers',
        'Internes'] else x)

VAR_EXPL_CAT_2 = VAR_EXPL_CAT_2.apply(lambda x: np.log(x) if x.name in ['Personnel médical salarié','Personnel médical libéral',
       'Personnels de direction et administratifs',
       'Personnels des services de soins', 'Personnels éducatifs et sociaux',
       'Personnels médico-techniques', 'Personnels techniques et ouvriers',
        'Internes'] else x)


In [110]:
VAR_CTRL.describe()

Points_ISA         stjr  Capital_Lit_MCO  Capital_Bloc  \
count  1.162000e+03  1162.000000      1139.000000   1153.000000   
mean   2.939915e+05     1.707401       125.109263      7.379577   
std    6.281782e+05     0.716795       127.684194      7.191520   
min    1.347900e+02     1.000000         0.000000      0.000000   
25%    2.993899e+04     1.000000        35.000000      1.000000   
50%    1.016205e+05     2.000000        85.000000      6.000000   
75%    2.975944e+05     2.000000       171.166667     10.500000   
max    9.116903e+06     3.000000      1055.000000     44.000000   

       Capital_imagerie  Capital_Examens_biologiques  Capital_telemed  \
count       1162.000000                 1.149000e+03      1162.000000   
mean           4.256971                 2.508881e+05         0.728148   
std            5.047103                 6.567670e+05         1.514649   
min            0.000000                 0.000000e+00         0.000000   
25%            0.000000                 0.000000e+00         0.000000   
50%            3.000000                 0.000000e+00         0.000000   
75%            7.000000                 6.261400e+04         1.000000   
max           28.000000                 1.060114e+07        11.000000   

          SEJHC_MED     SEJHC_CHI    SEJHC_OBS     SEJHC_MCO  \
count   1162.000000   1162.000000  1161.000000   1162.000000   
mean    3953.236728   2091.528256   673.093714   6717.279443   
std     5172.505167   2471.292213  1146.876402   7600.700172   
min        0.000000      0.000000     0.000000      0.000000   
25%      364.750000      0.000000     0.000000   1204.500000   
50%     1963.000000   1428.000000     0.000000   4168.500000   
75%     5581.250000   3168.312500  1081.000000   9749.750000   
max    39884.000000  20991.000000  7639.000000  60719.000000   

       Taux_chomage_par_dep    Population  Dep Moyenne Salaires Horaires  \
count           1162.000000  1.162000e+03                    1162.000000   
mean               8.290103  1.367009e+05                      11.793749   
std                1.660836  3.998901e+05                       1.908702   
min                5.000000  1.820000e+02                       9.862500   
25%                7.000000  8.053750e+03                      10.633333   
50%                8.000000  2.295500e+04                      11.204762   
75%                9.300000  6.171825e+04                      11.981429   
max               13.800000  2.125851e+06                      17.800000   

       Dep Moyenne Salaires Cadre Horaires  \
count                          1162.000000   
mean                             21.764363   
std                               1.674054   
min                              19.750000   
25%                              20.775000   
50%                              21.280808   
75%                              22.083750   
max                              27.600000   

       Dep Moyenne Salaires Prof Intermédiaire Horaires  \
count                                       1162.000000   
mean                                          12.674821   
std                                            0.523229   
min                                           11.740000   
25%                                           12.346721   
50%                                           12.535714   
75%                                           12.902500   
max                                           14.185000   

       Dep Moyenne Salaires Employé Horaires  \
count                            1162.000000   
mean                                8.783985   
std                                 0.353571   
min                                 8.261905   
25%                                 8.555172   
50%                                 8.700862   
75%                                 8.848113   
max                                 9.742553   

       Moyenne Revenus Fiscaux Départementaux  
count                             1162.000000  
mean     

On garde les 25% d'hopitaux avec les salaires horaires des cadres les plus bas (<20.7750, et les 25% d'hopitaux avec les salaires horaires les plus élevés.(>22.083750)

In [111]:
VAR_CTRL_2 = VAR_CTRL.copy()

VAR_CTRL_2 = VAR_CTRL_2.fillna(0)
VAR_CTRL_2 = VAR_CTRL_2.apply(lambda x: x+1 if x.name in ['Points_ISA', 'Capital_Lit_MCO', 'Capital_Bloc', 'Capital_imagerie',
       'Capital_Examens_biologiques', 'Capital_telemed', 'Population'] else x)

VAR_CTRL_2 = VAR_CTRL_2.apply(lambda x: np.log(x) if x.name in ['Points_ISA', 'Capital_Lit_MCO', 'Capital_Bloc', 'Capital_imagerie',
       'Capital_Examens_biologiques', 'Capital_telemed', 'Population'] else x)


In [112]:
# Les établissements avec des revenus départements  trés élevés

VAR_CTRL_ELEVES_2 = VAR_CTRL_2.copy()
VAR_CTRL_ELEVES_2= VAR_CTRL_ELEVES_2.rename(columns = {'Dep Moyenne Salaires Cadre Horaires': 'sal_cadre_moyen'})

VAR_CTRL_ELEVES_2= VAR_CTRL_ELEVES_2[VAR_CTRL_ELEVES_2.sal_cadre_moyen >= 21.280808]

In [113]:
# Les établissements avec des revenus départements trésfaibles

VAR_CTRL_FAIBLES_2 = VAR_CTRL_2.copy()
VAR_CTRL_FAIBLES_2 = VAR_CTRL_FAIBLES_2.rename(columns = {'Dep Moyenne Salaires Cadre Horaires': 'sal_cadre_moyen'})

VAR_CTRL_FAIBLES_2 = VAR_CTRL_FAIBLES_2[VAR_CTRL_FAIBLES_2.sal_cadre_moyen <= 21.280808]

In [114]:
analyse_sal_faibles = pd.merge(VAR_CTRL_FAIBLES_2, VAR_EXPL_CAT_2)

analyse_sal_eleves = pd.merge(VAR_CTRL_ELEVES_2, VAR_EXPL_CAT_2)

In [115]:
analyse_sal_faibles_X =analyse_sal_faibles[["Personnel médical salarié",'Personnel médical libéral',"Internes",'Personnels de direction et administratifs',
       'Personnels des services de soins', 'Personnels éducatifs et sociaux',
       'Personnels médico-techniques', 'Personnels techniques et ouvriers', 
        'Capital_Lit_MCO', 'Capital_Bloc', 'Capital_imagerie', 'Population',
       'Capital_Examens_biologiques', 'Capital_telemed', 'STJR_1','STJR_2','STJR_3']]
analyse_sal_faibles_Y =analyse_sal_faibles[['Points_ISA']]
analyse_sal_eleves_X =analyse_sal_eleves[["Personnel médical salarié",'Personnel médical libéral',"Internes",'Personnels de direction et administratifs',
       'Personnels des services de soins', 'Personnels éducatifs et sociaux',
       'Personnels médico-techniques', 'Personnels techniques et ouvriers', 
       'Capital_Lit_MCO', 'Capital_Bloc', 'Capital_imagerie', 'Population',
       'Capital_Examens_biologiques', 'Capital_telemed', 'STJR_1','STJR_2','STJR_3']]
analyse_dep_eleves_Y =analyse_sal_eleves[['Points_ISA']]

In [116]:
analyse_sal_faibles_X = sm.add_constant(analyse_sal_faibles_X) # on ajoute une constante, sinon la régression se fait sans, par défaut
y = analyse_sal_faibles[['Points_ISA']]

model_sal_faibles = sm.OLS(y, analyse_sal_faibles_X)
results_sal_faibles = model_sal_faibles.fit()
print(results_sal_faibles.summary())

                            OLS Regression Results                            
Dep. Variable:             Points_ISA   R-squared:                       0.897
Model:                            OLS   Adj. R-squared:                  0.894
Method:                 Least Squares   F-statistic:                     291.5
Date:                Wed, 19 May 2021   Prob (F-statistic):          6.64e-252
Time:                        15:35:24   Log-Likelihood:                -476.73
No. Observations:                 552   AIC:                             987.5
Df Residuals:                     535   BIC:                             1061.
Df Model:                          16                                         
Covariance Type:            nonrobust                                         
                                                coef    std err          t      P>|t|      [0.025      0.975]
----------------------------------------------------------------------------------------------------

In [117]:
analyse_sal_eleves_X = sm.add_constant(analyse_sal_eleves_X) # on ajoute une constante, sinon la régression se fait sans, par défaut
y = analyse_sal_eleves[['Points_ISA']]

model_sal_eleves = sm.OLS(y, analyse_sal_eleves_X)
results_sal_eleves = model_sal_eleves.fit()
print(results_sal_eleves.summary())

                            OLS Regression Results                            
Dep. Variable:             Points_ISA   R-squared:                       0.891
Model:                            OLS   Adj. R-squared:                  0.888
Method:                 Least Squares   F-statistic:                     298.4
Date:                Wed, 19 May 2021   Prob (F-statistic):          2.84e-268
Time:                        15:35:24   Log-Likelihood:                -509.70
No. Observations:                 600   AIC:                             1053.
Df Residuals:                     583   BIC:                             1128.
Df Model:                          16                                         
Covariance Type:            nonrobust                                         
                                                coef    std err          t      P>|t|      [0.025      0.975]
----------------------------------------------------------------------------------------------------

# Troisième analyse : effet du salaire moyen département des employés sur la participation des personnels des services de soins et des services techniques

In [118]:
VAR_EXPL_CAT_3 = VAR_EXPL_CAT.copy()

In [119]:
VAR_EXPL_CAT_3['Personnel médical salarié'] = VAR_EXPL_CAT_3['Médecins salariés'] + VAR_EXPL_CAT_3["Autres personnels médicaux salariés"]
VAR_EXPL_CAT_3['Personnel médical libéral'] = VAR_EXPL_CAT_3['Médecins libéraux'] + VAR_EXPL_CAT_3["Autres personnels médicaux libéraux"]

In [120]:
VAR_EXPL_CAT_3 = VAR_EXPL_CAT_3[['Numero_finess', 'Personnel médical salarié','Personnel médical libéral','stjr',
       'Personnels de direction et administratifs',
       'Personnels des services de soins', 'Personnels éducatifs et sociaux',
       'Personnels médico-techniques', 'Personnels techniques et ouvriers',
        'Internes', 'STJR_1','STJR_2','STJR_3']]

In [121]:
VAR_EXPL_CAT_3 = VAR_EXPL_CAT_3.fillna(0)

VAR_EXPL_CAT_3 = VAR_EXPL_CAT_3.apply(lambda x: x+1 if x.name in ['Personnel médical salarié','Personnel médical libéral',
       'Personnels de direction et administratifs',
       'Personnels des services de soins', 'Personnels éducatifs et sociaux',
       'Personnels médico-techniques', 'Personnels techniques et ouvriers',
        'Internes'] else x)

VAR_EXPL_CAT_3 = VAR_EXPL_CAT_3.apply(lambda x: np.log(x) if x.name in ['Personnel médical salarié','Personnel médical libéral',
       'Personnels de direction et administratifs',
       'Personnels des services de soins', 'Personnels éducatifs et sociaux',
       'Personnels médico-techniques', 'Personnels techniques et ouvriers',
        'Internes'] else x)

In [122]:
VAR_CTRL.describe()

Points_ISA         stjr  Capital_Lit_MCO  Capital_Bloc  \
count  1.162000e+03  1162.000000      1139.000000   1153.000000   
mean   2.939915e+05     1.707401       125.109263      7.379577   
std    6.281782e+05     0.716795       127.684194      7.191520   
min    1.347900e+02     1.000000         0.000000      0.000000   
25%    2.993899e+04     1.000000        35.000000      1.000000   
50%    1.016205e+05     2.000000        85.000000      6.000000   
75%    2.975944e+05     2.000000       171.166667     10.500000   
max    9.116903e+06     3.000000      1055.000000     44.000000   

       Capital_imagerie  Capital_Examens_biologiques  Capital_telemed  \
count       1162.000000                 1.149000e+03      1162.000000   
mean           4.256971                 2.508881e+05         0.728148   
std            5.047103                 6.567670e+05         1.514649   
min            0.000000                 0.000000e+00         0.000000   
25%            0.000000                 0.000000e+00         0.000000   
50%            3.000000                 0.000000e+00         0.000000   
75%            7.000000                 6.261400e+04         1.000000   
max           28.000000                 1.060114e+07        11.000000   

          SEJHC_MED     SEJHC_CHI    SEJHC_OBS     SEJHC_MCO  \
count   1162.000000   1162.000000  1161.000000   1162.000000   
mean    3953.236728   2091.528256   673.093714   6717.279443   
std     5172.505167   2471.292213  1146.876402   7600.700172   
min        0.000000      0.000000     0.000000      0.000000   
25%      364.750000      0.000000     0.000000   1204.500000   
50%     1963.000000   1428.000000     0.000000   4168.500000   
75%     5581.250000   3168.312500  1081.000000   9749.750000   
max    39884.000000  20991.000000  7639.000000  60719.000000   

       Taux_chomage_par_dep    Population  Dep Moyenne Salaires Horaires  \
count           1162.000000  1.162000e+03                    1162.000000   
mean               8.290103  1.367009e+05                      11.793749   
std                1.660836  3.998901e+05                       1.908702   
min                5.000000  1.820000e+02                       9.862500   
25%                7.000000  8.053750e+03                      10.633333   
50%                8.000000  2.295500e+04                      11.204762   
75%                9.300000  6.171825e+04                      11.981429   
max               13.800000  2.125851e+06                      17.800000   

       Dep Moyenne Salaires Cadre Horaires  \
count                          1162.000000   
mean                             21.764363   
std                               1.674054   
min                              19.750000   
25%                              20.775000   
50%                              21.280808   
75%                              22.083750   
max                              27.600000   

       Dep Moyenne Salaires Prof Intermédiaire Horaires  \
count                                       1162.000000   
mean                                          12.674821   
std                                            0.523229   
min                                           11.740000   
25%                                           12.346721   
50%                                           12.535714   
75%                                           12.902500   
max                                           14.185000   

       Dep Moyenne Salaires Employé Horaires  \
count                            1162.000000   
mean                                8.783985   
std                                 0.353571   
min                                 8.261905   
25%                                 8.555172   
50%                                 8.700862   
75%                                 8.848113   
max                                 9.742553   

       Moyenne Revenus Fiscaux Départementaux  
count                             1162.000000  
mean     

In [123]:
VAR_CTRL_3 = VAR_CTRL.copy()
VAR_CTRL_3= VAR_CTRL_3.fillna(0)
VAR_CTRL_3 = VAR_CTRL_3.apply(lambda x: x+1 if x.name in ['Points_ISA', 'Capital_Lit_MCO', 'Capital_Bloc', 'Capital_imagerie',
       'Capital_Examens_biologiques', 'Capital_telemed', 'Population'] else x)

VAR_CTRL_3 = VAR_CTRL_3.apply(lambda x: np.log(x) if x.name in ['Points_ISA', 'Capital_Lit_MCO', 'Capital_Bloc', 'Capital_imagerie',
       'Capital_Examens_biologiques', 'Capital_telemed', 'Population'] else x)

On garde les 25% d'hopitaux avec les salaires horaires des employés les plus bas (<8.555172, et les 25% d'hopitaux avec les salaires horaires les plus élevés.(>8.848113)

In [124]:
# Les établissements avec des revenus départements  trés élevés

VAR_CTRL_ELEVES_3 = VAR_CTRL_3.copy()
VAR_CTRL_ELEVES_3= VAR_CTRL_ELEVES_3.rename(columns = {'Dep Moyenne Salaires Employé Horaires': 'sal_employes_moyen'})

VAR_CTRL_ELEVES_3= VAR_CTRL_ELEVES_3[VAR_CTRL_ELEVES_3.sal_employes_moyen > 8.700862]

In [125]:
# Les établissements avec des revenus départements trésfaibles

VAR_CTRL_FAIBLES_3 = VAR_CTRL_3.copy()
VAR_CTRL_FAIBLES_3 = VAR_CTRL_FAIBLES_3.rename(columns = {'Dep Moyenne Salaires Employé Horaires': 'sal_employe_moyen'})

VAR_CTRL_FAIBLES_3 = VAR_CTRL_FAIBLES_3[VAR_CTRL_FAIBLES_3.sal_employe_moyen <= 8.700862]

In [126]:
analyse_sal_faibles_3 = pd.merge(VAR_CTRL_FAIBLES_3, VAR_EXPL_CAT_3)

analyse_sal_eleves_3 = pd.merge(VAR_CTRL_ELEVES_3, VAR_EXPL_CAT_3)

In [127]:
analyse_sal_faibles_X_3 =analyse_sal_faibles_3[["Personnel médical salarié",'Personnel médical libéral',"Internes",'Personnels de direction et administratifs',
       'Personnels des services de soins', 'Personnels éducatifs et sociaux',
       'Personnels médico-techniques', 'Personnels techniques et ouvriers', 
        'Capital_Lit_MCO', 'Capital_Bloc', 'Capital_imagerie', 'Population',
       'Capital_Examens_biologiques', 'Capital_telemed', 'STJR_1','STJR_2','STJR_3']]
analyse_sal_faibles_Y_3 =analyse_sal_faibles_3[['Points_ISA']]
analyse_sal_eleves_X_3 =analyse_sal_eleves_3[["Personnel médical salarié",'Personnel médical libéral',"Internes",'Personnels de direction et administratifs',
       'Personnels des services de soins', 'Personnels éducatifs et sociaux',
       'Personnels médico-techniques', 'Personnels techniques et ouvriers', 
       'Capital_Lit_MCO', 'Capital_Bloc', 'Capital_imagerie', 'Population',
       'Capital_Examens_biologiques', 'Capital_telemed', 'STJR_1','STJR_2','STJR_3']]
analyse_dep_eleves_Y_3=analyse_sal_eleves_3[['Points_ISA']]

In [128]:
analyse_sal_faibles_X_3

Personnel médical salarié  Personnel médical libéral  Internes  \
0                     2.109000                   3.960813  0.000000   
1                     0.000000                   4.135167  0.000000   
2                     3.835790                   3.540959  0.000000   
3                     0.000000                   0.000000  0.000000   
4                     5.860102                   0.000000  5.544884   
..                         ...                        ...       ...   
560                   3.270709                   0.000000  1.081805   
561                   3.951532                   2.140066  2.944439   
562                   0.641854                   1.386294  0.000000   
563                   0.223144                   2.014903  0.000000   
564                   0.698135                   1.386294  0.000000   

     Personnels de direction et administratifs  \
0                                     4.939354   
1                                     3.952589   
2                                     4.478018   
3                                     2.529721   
4                                     5.301836   
..                                         ...   
560                                   3.443618   
561                                   3.568593   
562                                   2.079442   
563                                   1.015231   
564                                   1.501853   

     Personnels des services de soins  Personnels éducatifs et sociaux  \
0                            5.901294                         0.000000   
1                            5.936058                         0.587787   
2                            5.876951                         0.883768   
3                            4.851640                         1.316408   
4                            7.051423                         3.069215   
..                                ...                              ...   
560                          5.170768                         1.098612   
561                          5.004617                         1.688249   
562                          3.858622                         0.693147   
563                          3.191299                         0.000000   
564                          3.677313                         0.165514   

     Personnels médico-techniques  Personnels techniques et ouvriers  \
0                        4.266756                           2.766948   
1                        2.399712                           1.637053   
2                        3.518980                           2.969902   
3                        0.693147                           1.558145   
4                        4.721485                           5.362957   
..                            ...                                ...   
560                      3.430756                           3.658420   
561                      2.976549                           3.140482   
562                      0.693147                           2.505526   
563                      0.223144                           1.560248   
564                      0.000000                           2.054124   

     Capital_Lit_MCO  Capital_Bloc  Capital_imagerie  Population  \
0           5.117994      2.833213          2.484907   11.609526   
1           5.318120      2.639057          0.000000   11.609526   
2           5.273000      2.639057          2.397895   11.083972   
3           2.944439      0.000000          0.693147   11.083972   
4           5.996452      3.806662          3.178054   11.083972   
..               ...           ...               ...         ...   
560         3.555348      1.791759          1.386294    8.822912   
561         4.219508      1.704748          1.945910    9.375261   
562         3.178054      0.000000          0.693147    8.038512   
563         3.135494      0.000000          0.000000    8.371011   
564         3.044522      0.000000          0.000

In [129]:
analyse_sal_faibles_X_3 = sm.add_constant(analyse_sal_faibles_X_3) # on ajoute une constante, sinon la régression se fait sans, par défaut
y = analyse_sal_faibles_3[['Points_ISA']]

model_sal_faibles_3 = sm.OLS(y, analyse_sal_faibles_X_3)
results_sal_faibles_3 = model_sal_faibles.fit()
print(results_sal_faibles_3.summary())

                            OLS Regression Results                            
Dep. Variable:             Points_ISA   R-squared:                       0.897
Model:                            OLS   Adj. R-squared:                  0.894
Method:                 Least Squares   F-statistic:                     291.5
Date:                Wed, 19 May 2021   Prob (F-statistic):          6.64e-252
Time:                        15:35:24   Log-Likelihood:                -476.73
No. Observations:                 552   AIC:                             987.5
Df Residuals:                     535   BIC:                             1061.
Df Model:                          16                                         
Covariance Type:            nonrobust                                         
                                                coef    std err          t      P>|t|      [0.025      0.975]
----------------------------------------------------------------------------------------------------

In [130]:
analyse_sal_eleves_X_3 = sm.add_constant(analyse_sal_eleves_X_3) # on ajoute une constante, sinon la régression se fait sans, par défaut
y = analyse_sal_eleves_3[['Points_ISA']]

model_sal_eleves_3 = sm.OLS(y, analyse_sal_eleves_X_3)
results_sal_eleves_3 = model_sal_eleves.fit()
print(results_sal_eleves_3.summary())

                            OLS Regression Results                            
Dep. Variable:             Points_ISA   R-squared:                       0.891
Model:                            OLS   Adj. R-squared:                  0.888
Method:                 Least Squares   F-statistic:                     298.4
Date:                Wed, 19 May 2021   Prob (F-statistic):          2.84e-268
Time:                        15:35:24   Log-Likelihood:                -509.70
No. Observations:                 600   AIC:                             1053.
Df Residuals:                     583   BIC:                             1128.
Df Model:                          16                                         
Covariance Type:            nonrobust                                         
                                                coef    std err          t      P>|t|      [0.025      0.975]
----------------------------------------------------------------------------------------------------

# Taille des villes : est ce la participation aux Points-ISA est la même dans les grandes et petites villes?

On prend les 25% des villes avec le moins d'ahbitants, par rapport aux 25% des villes avec le plus d'habitants

In [131]:
VAR_EXPL_CAT_4 = VAR_EXPL_CAT.copy()

In [132]:
VAR_EXPL_CAT_4 = VAR_EXPL_CAT_4[['Numero_finess', 'Personnel médical',
       'Personnels de direction et administratifs',
       'Personnels des services de soins', 'Personnels éducatifs et sociaux',
       'Personnels médico-techniques', 'Personnels techniques et ouvriers',
        'Internes', 'STJR_1','STJR_2','STJR_3']]

In [133]:
VAR_EXPL_CAT_4= VAR_EXPL_CAT_4.fillna(0)

VAR_EXPL_CAT_4 = VAR_EXPL_CAT_4.apply(lambda x: x+1 if x.name in ['Personnel médical',
       'Personnels de direction et administratifs',
       'Personnels des services de soins', 'Personnels éducatifs et sociaux',
       'Personnels médico-techniques', 'Personnels techniques et ouvriers',
        'Internes'] else x)

VAR_EXPL_CAT_4 = VAR_EXPL_CAT_4.apply(lambda x: np.log(x) if x.name in ['Personnel médical',
       'Personnels de direction et administratifs',
       'Personnels des services de soins', 'Personnels éducatifs et sociaux',
       'Personnels médico-techniques', 'Personnels techniques et ouvriers',
        'Internes'] else x)

In [134]:
VAR_CTRL.describe()

Points_ISA         stjr  Capital_Lit_MCO  Capital_Bloc  \
count  1.162000e+03  1162.000000      1139.000000   1153.000000   
mean   2.939915e+05     1.707401       125.109263      7.379577   
std    6.281782e+05     0.716795       127.684194      7.191520   
min    1.347900e+02     1.000000         0.000000      0.000000   
25%    2.993899e+04     1.000000        35.000000      1.000000   
50%    1.016205e+05     2.000000        85.000000      6.000000   
75%    2.975944e+05     2.000000       171.166667     10.500000   
max    9.116903e+06     3.000000      1055.000000     44.000000   

       Capital_imagerie  Capital_Examens_biologiques  Capital_telemed  \
count       1162.000000                 1.149000e+03      1162.000000   
mean           4.256971                 2.508881e+05         0.728148   
std            5.047103                 6.567670e+05         1.514649   
min            0.000000                 0.000000e+00         0.000000   
25%            0.000000                 0.000000e+00         0.000000   
50%            3.000000                 0.000000e+00         0.000000   
75%            7.000000                 6.261400e+04         1.000000   
max           28.000000                 1.060114e+07        11.000000   

          SEJHC_MED     SEJHC_CHI    SEJHC_OBS     SEJHC_MCO  \
count   1162.000000   1162.000000  1161.000000   1162.000000   
mean    3953.236728   2091.528256   673.093714   6717.279443   
std     5172.505167   2471.292213  1146.876402   7600.700172   
min        0.000000      0.000000     0.000000      0.000000   
25%      364.750000      0.000000     0.000000   1204.500000   
50%     1963.000000   1428.000000     0.000000   4168.500000   
75%     5581.250000   3168.312500  1081.000000   9749.750000   
max    39884.000000  20991.000000  7639.000000  60719.000000   

       Taux_chomage_par_dep    Population  Dep Moyenne Salaires Horaires  \
count           1162.000000  1.162000e+03                    1162.000000   
mean               8.290103  1.367009e+05                      11.793749   
std                1.660836  3.998901e+05                       1.908702   
min                5.000000  1.820000e+02                       9.862500   
25%                7.000000  8.053750e+03                      10.633333   
50%                8.000000  2.295500e+04                      11.204762   
75%                9.300000  6.171825e+04                      11.981429   
max               13.800000  2.125851e+06                      17.800000   

       Dep Moyenne Salaires Cadre Horaires  \
count                          1162.000000   
mean                             21.764363   
std                               1.674054   
min                              19.750000   
25%                              20.775000   
50%                              21.280808   
75%                              22.083750   
max                              27.600000   

       Dep Moyenne Salaires Prof Intermédiaire Horaires  \
count                                       1162.000000   
mean                                          12.674821   
std                                            0.523229   
min                                           11.740000   
25%                                           12.346721   
50%                                           12.535714   
75%                                           12.902500   
max                                           14.185000   

       Dep Moyenne Salaires Employé Horaires  \
count                            1162.000000   
mean                                8.783985   
std                                 0.353571   
min                                 8.261905   
25%                                 8.555172   
50%                                 8.700862   
75%                                 8.848113   
max                                 9.742553   

       Moyenne Revenus Fiscaux Départementaux  
count                             1162.000000  
mean     

Les 25% des villes avec le moins d'habitants ont une population inférieure 8.055250e+03, tandis que les 25% des vilels avec le plus d'habitants, ont plus de 6.227500e+04 habitants

In [135]:
VAR_CTRL_4 = VAR_CTRL.copy()
VAR_CTRL_4= VAR_CTRL_4.fillna(0)

In [136]:
# Les établissements avec des revenus départements  trés élevés

VAR_CTRL_ELEVES_4 = VAR_CTRL_4.copy()

VAR_CTRL_ELEVES_4= VAR_CTRL_ELEVES_4[VAR_CTRL_ELEVES_4.Population >= 2.295500e+04]

In [137]:
VAR_CTRL_ELEVES_4 = VAR_CTRL_ELEVES_4.apply(lambda x: x+1 if x.name in ['Points_ISA', 'Capital_Lit_MCO', 'Capital_Bloc', 'Capital_imagerie',
       'Capital_Examens_biologiques', 'Capital_telemed', 'Population'] else x)

VAR_CTRL_ELEVES_4 = VAR_CTRL_ELEVES_4.apply(lambda x: np.log(x) if x.name in ['Points_ISA', 'Capital_Lit_MCO', 'Capital_Bloc', 'Capital_imagerie',
       'Capital_Examens_biologiques', 'Capital_telemed', 'Population'] else x)

In [138]:
# Les établissements avec des revenus départements trésfaibles

VAR_CTRL_FAIBLES_4 = VAR_CTRL_4.copy()
VAR_CTRL_FAIBLES_4 = VAR_CTRL_FAIBLES_4[VAR_CTRL_FAIBLES_4.Population <= 2.295500e+04]

In [139]:
VAR_CTRL_FAIBLES_4 = VAR_CTRL_FAIBLES_4.apply(lambda x: x+1 if x.name in ['Points_ISA', 'Capital_Lit_MCO', 'Capital_Bloc', 'Capital_imagerie',
       'Capital_Examens_biologiques', 'Capital_telemed', 'Population'] else x)


VAR_CTRL_FAIBLES_4 = VAR_CTRL_FAIBLES_4.apply(lambda x: np.log(x) if x.name in ['Points_ISA', 'Capital_Lit_MCO', 'Capital_Bloc', 'Capital_imagerie',
       'Capital_Examens_biologiques', 'Capital_telemed', 'Population'] else x)

In [140]:
analyse_pop_faibles_4 = pd.merge(VAR_CTRL_FAIBLES_4, VAR_EXPL_CAT_4)

analyse_pop_eleves_4 = pd.merge(VAR_CTRL_ELEVES_4, VAR_EXPL_CAT_4)

In [141]:
analyse_pop_eleves_4 

Numero_finess  Points_ISA  stjr dep            Departement COMINSEE  \
0       060000528   12.785191     3  06        Alpes-Maritimes    06088   
1       060780947   12.069987     3  06        Alpes-Maritimes    06088   
2       060791811   11.894328     3  06        Alpes-Maritimes    06088   
3       060780442   10.036417     2  06        Alpes-Maritimes    06088   
4       060780715   12.742719     2  06        Alpes-Maritimes    06088   
..            ...         ...   ...  ..                    ...      ...   
569     610780082   13.140031     1  61                   Orne    61001   
570     850000118   12.089712     2  85                 Vendée    85191   
571     850000019   14.232458     1  85                 Vendée    85191   
572     900000035   11.126500     2  90  Territoire de Belfort    90010   
573     120780044   13.518522     1  12                Aveyron    12202   

     Capital_Lit_MCO  Capital_Bloc  Capital_imagerie  \
0           4.859812      2.197225          2.302585   
1           4.663439      1.791759          1.609438   
2           4.521789      1.098612          1.098612   
3           3.433987      1.945910          0.693147   
4           5.549076      3.295837          2.397895   
..               ...           ...               ...   
569         5.007296      2.079442          2.251292   
570         4.709530      2.484907          0.000000   
571         5.546479      3.135494          2.512306   
572         3.871201      2.302585          0.000000   
573         5.736572      2.708050          2.397895   

     Capital_Examens_biologiques  ...  Personnel médical  \
0                       6.813445  ...           4.523201   
1                       0.000000  ...           4.302713   
2                       0.000000  ...           3.366261   
3                       0.000000  ...           3.496508   
4                       0.000000  ...           4.852030   
..                           ...  ...                ...   
569                    13.689514  ...           4.649315   
570                     0.000000  ...           3.645450   
571                    13.712304  ...           5.042887   
572                     0.000000  ...           3.100092   
573                    13.767213  ...           4.888518   

     Personnels de direction et administratifs  \
0                                     4.990160   
1                                     4.730569   
2                                     3.842244   
3                                     2.890372   
4                                     4.435686   
..                                         ...   
569                                   4.035361   
570                                   3.318178   
571                                   4.899629   
572                                   3.077312   
573                                   4.148201   

     Personnels des services de soins  Personnels éducatifs et sociaux  \
0                            5.777807                         1.710188   
1                            5.863290                         1.178655   
2                            5.509995                         1.523880   
3                            4.333361                         0.693147   
4                            6.043583                         0.553885   
..                                ...                              ...   
569                          5.783805                         1.342865   
570                          5.194789                         0.587787   
571                          6.668893                         2.253745   
572                          4.759092                         0.693147   
573                          5.691541                         2.412933   

     Personnels médico-techniques  Personnels techniques et ouvriers  \
0                        5.112831                           3.697344   
1                        3.119718                           3.006178   
2  

In [142]:
analyse_pop_faibles_X_4 =analyse_pop_faibles_4[["Personnel médical","Internes",'Personnels de direction et administratifs',
       'Personnels des services de soins', 'Personnels éducatifs et sociaux',
       'Personnels médico-techniques', 'Personnels techniques et ouvriers', 
        'Capital_Lit_MCO', 'Capital_Bloc', 'Capital_imagerie',
       'Capital_Examens_biologiques', 'Capital_telemed', 'STJR_1','STJR_2','STJR_3',]]
analyse_pop_faibles_Y_4 =analyse_pop_faibles_4[['Points_ISA']]
analyse_pop_eleves_X_4 =analyse_pop_eleves_4[["Personnel médical","Internes",'Personnels de direction et administratifs',
       'Personnels des services de soins', 'Personnels éducatifs et sociaux',
       'Personnels médico-techniques', 'Personnels techniques et ouvriers', 
       'Capital_Lit_MCO', 'Capital_Bloc', 'Capital_imagerie',
       'Capital_Examens_biologiques', 'Capital_telemed', 'STJR_1','STJR_2','STJR_3',]]
analyse_pop_eleves_Y_4=analyse_pop_eleves_4[['Points_ISA']]

In [143]:
analyse_pop_faibles_X_4 = sm.add_constant(analyse_pop_faibles_X_4) # on ajoute une constante, sinon la régression se fait sans, par défaut
y = analyse_pop_faibles_4[['Points_ISA']]

model_pop_faibles_4 = sm.OLS(y, analyse_pop_faibles_X_4)
results_pop_faibles_4 = model_pop_faibles_4.fit()
print(results_pop_faibles_4.summary())

                            OLS Regression Results                            
Dep. Variable:             Points_ISA   R-squared:                       0.877
Model:                            OLS   Adj. R-squared:                  0.874
Method:                 Least Squares   F-statistic:                     288.0
Date:                Wed, 19 May 2021   Prob (F-statistic):          7.22e-246
Time:                        15:35:25   Log-Likelihood:                -528.51
No. Observations:                 578   AIC:                             1087.
Df Residuals:                     563   BIC:                             1152.
Df Model:                          14                                         
Covariance Type:            nonrobust                                         
                                                coef    std err          t      P>|t|      [0.025      0.975]
----------------------------------------------------------------------------------------------------

In [144]:
analyse_pop_eleves_X_4 = sm.add_constant(analyse_pop_eleves_X_4) # on ajoute une constante, sinon la régression se fait sans, par défaut
y = analyse_pop_eleves_4[['Points_ISA']]

model_pop_eleves_4 = sm.OLS(y, analyse_pop_eleves_X_4)
results_pop_eleves_4 = model_pop_eleves_4.fit()
print(results_pop_eleves_4.summary())

                            OLS Regression Results                            
Dep. Variable:             Points_ISA   R-squared:                       0.902
Model:                            OLS   Adj. R-squared:                  0.899
Method:                 Least Squares   F-statistic:                     366.5
Date:                Wed, 19 May 2021   Prob (F-statistic):          8.41e-271
Time:                        15:35:25   Log-Likelihood:                -410.70
No. Observations:                 574   AIC:                             851.4
Df Residuals:                     559   BIC:                             916.7
Df Model:                          14                                         
Covariance Type:            nonrobust                                         
                                                coef    std err          t      P>|t|      [0.025      0.975]
----------------------------------------------------------------------------------------------------

# En fonction de la majoritée politique du département

In [145]:
VAR_EXPL_CAT_5 = VAR_EXPL_CAT.copy()

In [146]:
VAR_EXPL_CAT_5 = VAR_EXPL_CAT_5[['Numero_finess', 'Personnel médical','STJR_1','STJR_2','STJR_3',
       'Personnels de direction et administratifs',
       'Personnels des services de soins', 'Personnels éducatifs et sociaux',
       'Personnels médico-techniques', 'Personnels techniques et ouvriers',
        'Internes']]
VAR_EXPL_CAT_5 = VAR_EXPL_CAT_5.fillna(0)

In [147]:
VAR_EXPL_CAT_5 = VAR_EXPL_CAT_5.apply(lambda x: x+1 if x.name in ['Personnel médical',
       'Personnels de direction et administratifs',
       'Personnels des services de soins', 'Personnels éducatifs et sociaux',
       'Personnels médico-techniques', 'Personnels techniques et ouvriers',
        'Internes'] else x)


VAR_EXPL_CAT_5 = VAR_EXPL_CAT_5.apply(lambda x: np.log(x) if x.name in ['Personnel médical',
       'Personnels de direction et administratifs',
       'Personnels des services de soins', 'Personnels éducatifs et sociaux',
       'Personnels médico-techniques', 'Personnels techniques et ouvriers',
        'Internes'] else x)

In [148]:
VAR_CTRL.groupby("Bloc_politique").count()

Numero_finess  Points_ISA  stjr  dep  \
Bloc_politique                                                       
Centre droit à droite                   146         146   146  146   
Centre gauche à centre droit            556         556   556  556   
Droite à extrême droite                 287         287   287  287   
Extrême gauche à gauche                 173         173   173  173   

                              Departement  COMINSEE  Capital_Lit_MCO  \
Bloc_politique                                                         
Centre droit à droite                 146       146              144   
Centre gauche à centre droit          556       556              541   
Droite à extrême droite               287       287              282   
Extrême gauche à gauche               173       173              172   

                              Capital_Bloc  Capital_imagerie  \
Bloc_politique                                                 
Centre droit à droite                  145               146   
Centre gauche à centre droit           552               556   
Droite à extrême droite                283               287   
Extrême gauche à gauche                173               173   

                              Capital_Examens_biologiques  ...  SEJHC_CHI  \
Bloc_politique                                             ...              
Centre droit à droite                                 145  ...        146   
Centre gauche à centre droit                          550  ...        556   
Droite à extrême droite                               283  ...        287   
Extrême gauche à gauche                               171  ...        173   

                              SEJHC_OBS  SEJHC_MCO  Taux_chomage_par_dep  \
Bloc_politique                                                             
Centre droit à droite               146        146                   146   
Centre gauche à centre droit        555        556                   556   
Droite à extrême droite             287        287                   287   
Extrême gauche à gauche             173        173                   173   

                              Population  Dep Moyenne Salaires Horaires  \
Bloc_politique                                                            
Centre droit à droite                146                            146   
Centre gauche à centre droit         556                            556   
Droite à extrême droite              287                            287   
Extrême gauche à gauche              173                            173   

                              Dep Moyenne Salaires Cadre Horaires  \
Bloc_politique                                                      
Centre droit à droite                                         146   
Centre gauche à centre droit                                  556   
Droite à extrême droite                                       287   
Extrême gauche à gauche                                       173   

                              Dep Moyenne Salaires Prof Intermédiaire Horaires  \
Bloc_politique                                                                   
Centre droit à droite                                                      146   
Centre gauche à centre droit                                               556   
Droite à extrême droite                                                    287   
Extrême gauche à gauche                                                    173   

                              Dep Moyenne Salaires Employé Horaires  \
Bloc_politique                                                        
Centre droit à droite                                           146   
Centre gauche à centre droit                                    556   
Droite à extrême droite                                         287   
Extrême gauche à gauche                                         173   

                              Moyenne Revenus Fiscaux Départementaux  
Bloc_politique

In [149]:
# Les établissements avec des revenus départements  trés élevés

VAR_CTRL_GAUCHE_5 = VAR_CTRL.copy()
VAR_CTRL_GAUCHE_5= VAR_CTRL_GAUCHE_5.fillna(0)

VAR_CTRL_GAUCHE_5 = VAR_CTRL_GAUCHE_5.apply(lambda x: x+1 if x.name in ['Points_ISA', 'Capital_Lit_MCO', 'Capital_Bloc', 'Capital_imagerie',
       'Capital_Examens_biologiques', 'Capital_telemed', 'Population'] else x)

VAR_CTRL_GAUCHE_5 = VAR_CTRL_GAUCHE_5.apply(lambda x: np.log(x) if x.name in ['Points_ISA', 'Capital_Lit_MCO', 'Capital_Bloc', 'Capital_imagerie',
       'Capital_Examens_biologiques', 'Capital_telemed', 'Population'] else x)

VAR_CTRL_GAUCHE_5_1= VAR_CTRL_GAUCHE_5[VAR_CTRL_GAUCHE_5.Bloc_politique == "Centre gauche à centre droit"]
VAR_CTRL_GAUCHE_5_2 = VAR_CTRL_GAUCHE_5[VAR_CTRL_GAUCHE_5.Bloc_politique == "Extrême gauche à gauche"]
VAR_CTRL_GAUCHE_5 = pd.concat([VAR_CTRL_GAUCHE_5_1,VAR_CTRL_GAUCHE_5_2])

In [150]:
VAR_CTRL_DROITE_5 = VAR_CTRL.copy()

VAR_CTRL_DROITE_5= VAR_CTRL_DROITE_5.fillna(0)
VAR_CTRL_DROITE_5 = VAR_CTRL_DROITE_5.apply(lambda x: x+1 if x.name in ['Points_ISA', 'Capital_Lit_MCO', 'Capital_Bloc', 'Capital_imagerie',
       'Capital_Examens_biologiques', 'Capital_telemed', 'Population'] else x)

VAR_CTRL_DROITE_5 = VAR_CTRL_DROITE_5.apply(lambda x: np.log(x) if x.name in ['Points_ISA', 'Capital_Lit_MCO', 'Capital_Bloc', 'Capital_imagerie',
       'Capital_Examens_biologiques', 'Capital_telemed', 'Population'] else x)


VAR_CTRL_DROITE_5_1= VAR_CTRL_DROITE_5[VAR_CTRL_DROITE_5.Bloc_politique == "Centre droit à droite"]
VAR_CTRL_DROITE_5_2 = VAR_CTRL_DROITE_5[VAR_CTRL_DROITE_5.Bloc_politique == "Droite à extrême droite"]
VAR_CTRL_DROITE_5 = pd.concat([VAR_CTRL_DROITE_5_1,VAR_CTRL_DROITE_5_2])

In [151]:
analyse_dep_gauche_5 = pd.merge(VAR_CTRL_GAUCHE_5, VAR_EXPL_CAT_5)

analyse_dep_droite_5 = pd.merge(VAR_CTRL_DROITE_5, VAR_EXPL_CAT_5)

In [152]:
analyse_dep_gauche_X_5 =analyse_dep_gauche_5[["Personnel médical","Internes",'Personnels de direction et administratifs',
       'Personnels des services de soins', 'Personnels éducatifs et sociaux',
       'Personnels médico-techniques', 'Personnels techniques et ouvriers', 
        'Capital_Lit_MCO', 'Capital_Bloc', 'Capital_imagerie',
       'Capital_Examens_biologiques', 'Capital_telemed', 'STJR_1', 'STJR_2','STJR_3']]
analyse_dep_gauche_Y_5 =analyse_dep_gauche_5[['Points_ISA']]
analyse_dep_droite_X_5 =analyse_dep_droite_5[["Personnel médical","Internes",'Personnels de direction et administratifs',
       'Personnels des services de soins', 'Personnels éducatifs et sociaux',
       'Personnels médico-techniques', 'Personnels techniques et ouvriers', 
       'Capital_Lit_MCO', 'Capital_Bloc', 'Capital_imagerie',
       'Capital_Examens_biologiques', 'Capital_telemed', 'STJR_1', 'STJR_2','STJR_3']]
analyse_dep_droite_Y_5 =analyse_dep_droite_5[['Points_ISA']]

In [153]:
analyse_dep_gauche_X_5 = sm.add_constant(analyse_dep_gauche_X_5) # on ajoute une constante, sinon la régression se fait sans, par défaut
y = analyse_dep_gauche_5[['Points_ISA']]

model_dep_gauche_5 = sm.OLS(y, analyse_dep_gauche_X_5)
results_dep_gauche_5 = model_dep_gauche_5.fit()
print(results_dep_gauche_5.summary())

                            OLS Regression Results                            
Dep. Variable:             Points_ISA   R-squared:                       0.905
Model:                            OLS   Adj. R-squared:                  0.904
Method:                 Least Squares   F-statistic:                     483.5
Date:                Wed, 19 May 2021   Prob (F-statistic):               0.00
Time:                        15:35:25   Log-Likelihood:                -584.06
No. Observations:                 722   AIC:                             1198.
Df Residuals:                     707   BIC:                             1267.
Df Model:                          14                                         
Covariance Type:            nonrobust                                         
                                                coef    std err          t      P>|t|      [0.025      0.975]
----------------------------------------------------------------------------------------------------

In [154]:
analyse_dep_droite_X_5 = sm.add_constant(analyse_dep_droite_X_5) # on ajoute une constante, sinon la régression se fait sans, par défaut
y = analyse_dep_droite_5[['Points_ISA']]

model_dep_droite_5 = sm.OLS(y, analyse_dep_droite_X_5)
results_dep_droite_5 = model_dep_droite_5.fit()
print(results_dep_droite_5.summary())

                            OLS Regression Results                            
Dep. Variable:             Points_ISA   R-squared:                       0.877
Model:                            OLS   Adj. R-squared:                  0.873
Method:                 Least Squares   F-statistic:                     211.8
Date:                Wed, 19 May 2021   Prob (F-statistic):          5.35e-179
Time:                        15:35:25   Log-Likelihood:                -387.95
No. Observations:                 430   AIC:                             805.9
Df Residuals:                     415   BIC:                             866.9
Df Model:                          14                                         
Covariance Type:            nonrobust                                         
                                                coef    std err          t      P>|t|      [0.025      0.975]
----------------------------------------------------------------------------------------------------

# En séparant gauche, droite et centre

In [155]:
# Les établissements avec des revenus départements  trés élevés

VAR_CTRL_GAUCHE_5 = VAR_CTRL.copy()
VAR_CTRL_GAUCHE_5= VAR_CTRL_GAUCHE_5.fillna(0)
VAR_CTRL_GAUCHE_5 = VAR_CTRL_GAUCHE_5.apply(lambda x: x+1 if x.name in ['Points_ISA', 'Capital_Lit_MCO', 'Capital_Bloc', 'Capital_imagerie',
       'Capital_Examens_biologiques', 'Capital_telemed', 'Population'] else x)

VAR_CTRL_GAUCHE_5 = VAR_CTRL_GAUCHE_5.apply(lambda x: np.log(x) if x.name in ['Points_ISA', 'Capital_Lit_MCO', 'Capital_Bloc', 'Capital_imagerie',
       'Capital_Examens_biologiques', 'Capital_telemed', 'Population'] else x)

VAR_CTRL_GAUCHE_5 = VAR_CTRL_GAUCHE_5[VAR_CTRL_GAUCHE_5.Bloc_politique == "Extrême gauche à gauche"]

In [156]:
VAR_CTRL_CENTRE_5 = VAR_CTRL.copy()
VAR_CTRL_CENTRE_5= VAR_CTRL_CENTRE_5.fillna(0)
VAR_CTRL_CENTRE_5 = VAR_CTRL_CENTRE_5.apply(lambda x: x+1 if x.name in ['Points_ISA', 'Capital_Lit_MCO', 'Capital_Bloc', 'Capital_imagerie',
       'Capital_Examens_biologiques', 'Capital_telemed', 'Population'] else x)

VAR_CTRL_CENTRE_5 = VAR_CTRL_CENTRE_5.apply(lambda x: np.log(x) if x.name in ['Points_ISA', 'Capital_Lit_MCO', 'Capital_Bloc', 'Capital_imagerie',
       'Capital_Examens_biologiques', 'Capital_telemed', 'Population'] else x)


VAR_CTRL_CENTRE_5= VAR_CTRL_CENTRE_5[VAR_CTRL_CENTRE_5.Bloc_politique == "Centre gauche à centre droit"]

In [157]:
VAR_CTRL_DROITE_5 = VAR_CTRL.copy()
VAR_CTRL_DROITE_5= VAR_CTRL_DROITE_5.fillna(0)
VAR_CTRL_DROITE_5 = VAR_CTRL_DROITE_5.apply(lambda x: x+1 if x.name in ['Points_ISA', 'Capital_Lit_MCO', 'Capital_Bloc', 'Capital_imagerie',
       'Capital_Examens_biologiques', 'Capital_telemed', 'Population'] else x)

VAR_CTRL_DROITE_5 = VAR_CTRL_DROITE_5.apply(lambda x: np.log(x) if x.name in ['Points_ISA', 'Capital_Lit_MCO', 'Capital_Bloc', 'Capital_imagerie',
       'Capital_Examens_biologiques', 'Capital_telemed', 'Population'] else x)

VAR_CTRL_DROITE_5_1= VAR_CTRL_DROITE_5[VAR_CTRL_DROITE_5.Bloc_politique == "Centre droit à droite"]
VAR_CTRL_DROITE_5_2 = VAR_CTRL_DROITE_5[VAR_CTRL_DROITE_5.Bloc_politique == "Droite à extrême droite"]
VAR_CTRL_DROITE_5 = pd.concat([VAR_CTRL_DROITE_5_1,VAR_CTRL_DROITE_5_2])

In [158]:
VAR_CTRL_DROITE_5= VAR_CTRL_DROITE_5.fillna(0)
VAR_CTRL_GAUCHE_5 = VAR_CTRL_GAUCHE_5.fillna(0)
VAR_CTRL_CENTRE_5 = VAR_CTRL_CENTRE_5.fillna(0)
VAR_EXPL_CAT_5 = VAR_EXPL_CAT_5.fillna(0)
#stjr_1 = stjr_1.fillna(0)

In [159]:
analyse_dep_gauche_5 = pd.merge(VAR_CTRL_GAUCHE_5, VAR_EXPL_CAT_5)
analyse_dep_centre_5 = pd.merge(VAR_CTRL_CENTRE_5, VAR_EXPL_CAT_5)

analyse_dep_droite_5 = pd.merge(VAR_CTRL_DROITE_5, VAR_EXPL_CAT_5)

In [160]:
analyse_dep_gauche_X_5 =analyse_dep_gauche_5[["Personnel médical","Internes",'Personnels de direction et administratifs',
       'Personnels des services de soins', 'Personnels éducatifs et sociaux',
       'Personnels médico-techniques', 'Personnels techniques et ouvriers', 
        'Capital_Lit_MCO', 'Capital_Bloc', 'Capital_imagerie',
       'Capital_Examens_biologiques', 'Capital_telemed', 'STJR_1', 'STJR_2','STJR_3']]
analyse_dep_gauche_Y_5 =analyse_dep_gauche_5[['Points_ISA']]
analyse_dep_droite_X_5 =analyse_dep_droite_5[["Personnel médical","Internes",'Personnels de direction et administratifs',
       'Personnels des services de soins', 'Personnels éducatifs et sociaux',
       'Personnels médico-techniques', 'Personnels techniques et ouvriers', 
       'Capital_Lit_MCO', 'Capital_Bloc', 'Capital_imagerie',
       'Capital_Examens_biologiques', 'Capital_telemed', 'STJR_1', 'STJR_2','STJR_3']]
analyse_dep_droite_Y_5 =analyse_dep_droite_5[['Points_ISA']]
analyse_dep_centre_X_5 =analyse_dep_centre_5[["Personnel médical","Internes",'Personnels de direction et administratifs',
       'Personnels des services de soins', 'Personnels éducatifs et sociaux',
       'Personnels médico-techniques', 'Personnels techniques et ouvriers', 
        'Capital_Lit_MCO', 'Capital_Bloc', 'Capital_imagerie',
       'Capital_Examens_biologiques', 'Capital_telemed', 'STJR_1', 'STJR_2','STJR_3']]
analyse_dep_centre_Y_5 =analyse_dep_centre_5[['Points_ISA']]

In [161]:
analyse_dep_droite_X_5 = sm.add_constant(analyse_dep_droite_X_5) # on ajoute une constante, sinon la régression se fait sans, par défaut
y = analyse_dep_droite_5[['Points_ISA']]

model_dep_droite_5 = sm.OLS(y, analyse_dep_droite_X_5)
results_dep_droite_5 = model_dep_droite_5.fit()
print(results_dep_droite_5.summary())

                            OLS Regression Results                            
Dep. Variable:             Points_ISA   R-squared:                       0.877
Model:                            OLS   Adj. R-squared:                  0.873
Method:                 Least Squares   F-statistic:                     211.8
Date:                Wed, 19 May 2021   Prob (F-statistic):          5.35e-179
Time:                        15:35:26   Log-Likelihood:                -387.95
No. Observations:                 430   AIC:                             805.9
Df Residuals:                     415   BIC:                             866.9
Df Model:                          14                                         
Covariance Type:            nonrobust                                         
                                                coef    std err          t      P>|t|      [0.025      0.975]
----------------------------------------------------------------------------------------------------

In [162]:
analyse_dep_gauche_X_5 = sm.add_constant(analyse_dep_gauche_X_5) # on ajoute une constante, sinon la régression se fait sans, par défaut
y = analyse_dep_gauche_5[['Points_ISA']]

model_dep_gauche_5 = sm.OLS(y, analyse_dep_gauche_X_5)
results_dep_gauche_5 = model_dep_gauche_5.fit()
print(results_dep_gauche_5.summary())

                            OLS Regression Results                            
Dep. Variable:             Points_ISA   R-squared:                       0.923
Model:                            OLS   Adj. R-squared:                  0.917
Method:                 Least Squares   F-statistic:                     136.2
Date:                Wed, 19 May 2021   Prob (F-statistic):           1.79e-80
Time:                        15:35:26   Log-Likelihood:                -123.43
No. Observations:                 173   AIC:                             276.9
Df Residuals:                     158   BIC:                             324.1
Df Model:                          14                                         
Covariance Type:            nonrobust                                         
                                                coef    std err          t      P>|t|      [0.025      0.975]
----------------------------------------------------------------------------------------------------

In [163]:
analyse_dep_centre_X_5 = sm.add_constant(analyse_dep_centre_X_5) # on ajoute une constante, sinon la régression se fait sans, par défaut
y = analyse_dep_centre_5[['Points_ISA']]

model_dep_centre_5 = sm.OLS(y, analyse_dep_centre_X_5)
results_dep_centre_5 = model_dep_centre_5.fit()
print(results_dep_centre_5.summary())

                            OLS Regression Results                            
Dep. Variable:             Points_ISA   R-squared:                       0.903
Model:                            OLS   Adj. R-squared:                  0.900
Method:                 Least Squares   F-statistic:                     355.2
Date:                Wed, 19 May 2021   Prob (F-statistic):          8.25e-260
Time:                        15:35:26   Log-Likelihood:                -449.07
No. Observations:                 549   AIC:                             928.1
Df Residuals:                     534   BIC:                             992.8
Df Model:                          14                                         
Covariance Type:            nonrobust                                         
                                                coef    std err          t      P>|t|      [0.025      0.975]
----------------------------------------------------------------------------------------------------